src/moonlink/src/observability.rs

```diff
@@ -0,0 +1,25 @@
+use opentelemetry::metrics::Histogram;
+use opentelemetry::{global, KeyValue};
+use std::sync::Arc;
+
+pub struct SnapshotCreationStats {
+    latency_hist: Histogram<f64>,
+}
+
+impl SnapshotCreationStats {
+    pub fn new() -> Arc<Self> {
+        let meter = global::meter("snapshot_creation");
+        Arc::new(SnapshotCreationStats {
+            latency_hist: meter
+                .f64_histogram("snapshot_creation_latency")
+                .with_description("snapshot create latency histogram (second)")
+                .with_boundaries(vec![0.0, 0.2, 0.5, 0.8, 1.0, 2.0, 5.0])
```

Comment:
The distribution doesn't seem reasonable to me, we take mooncake snapshot every 500 milliseconds
https://github.com/Mooncake-Labs/moonlink/blob/65362866794a4230259a1695d5a6a8bcd8818b38/src/moonlink/src/table_handler_timer.rs#L19
which means in theory, all snapshot operation finishes within 500 millisec.